referenceframe/frame.go

```diff
@@ -445,3 +445,133 @@ func (rf rotationalFrame) MarshalJSON() ([]byte, error) {
+	orientationVector := spatial.OrientationVector{
+		OX:    math.Inf(1),
+		OY:    math.Inf(1),
+		OZ:    math.Inf(1),
+		Theta: math.Inf(1),
+	}
+	orientationVector.Normalize()
+	limits := []Limit{
+		{Min: math.Inf(-1), Max: math.Inf(1)},                         // X
+		{Min: math.Inf(-1), Max: math.Inf(1)},                         // Y
+		{Min: math.Inf(-1), Max: math.Inf(1)},                         // Z
+		{Min: -orientationVector.OX, Max: orientationVector.OX},       // OX
+		{Min: -orientationVector.OY, Max: orientationVector.OY},       // OY
+		{Min: -orientationVector.OZ, Max: orientationVector.OZ},       // OZ
+		{Min: -orientationVector.Theta, Max: orientationVector.Theta}, // Theta
+	}
+	return &poseFrame{
+		&baseFrame{name, limits},
+		geometry,
+	}, nil
+}
+
+// Transform on the poseFrame acts as the identity function. Whatever inputs are given are directly translated
+// in a 7DoF pose. We note that theta should be in radians.
+func (pf *poseFrame) Transform(inputs []Input) (spatial.Pose, error) {
+	if err := pf.baseFrame.validInputs(inputs); err != nil {
+		return nil, NewIncorrectInputLengthError(len(inputs), 7)
+	}
+	return inputsToPose(inputs), nil
+}
+
+// Interpolate interpolates the given amount between the two sets of inputs.
+func (pf *poseFrame) Interpolate(from, to []Input, by float64) ([]Input, error) {
+	if err := pf.baseFrame.validInputs(from); err != nil {
+		return nil, NewIncorrectInputLengthError(len(from), len(pf.DoF()))
+	}
+	if err := pf.baseFrame.validInputs(to); err != nil {
+		return nil, NewIncorrectInputLengthError(len(to), len(pf.DoF()))
+	}
+	fromPose, err := pf.Transform(from)
+	if err != nil {
+		return nil, err
+	}
+	toPose, err := pf.Transform(to)
+	if err != nil {
+		return nil, err
+	}
+	interpolatedPose := spatial.Interpolate(fromPose, toPose, by)
+	return PoseToInputsRadians(interpolatedPose), nil
+}
+
+// Geometries returns an object representing the 3D space associeted with the staticFrame.
+func (pf *poseFrame) Geometries(inputs []Input) (*GeometriesInFrame, error) {
+	transformByPose, err := pf.Transform(inputs)
+	if err != nil {
+		return nil, err
+	}
+	if len(pf.geometries) == 0 {
+		return NewGeometriesInFrame(pf.name, []spatial.Geometry{}), nil
+	}
+	transformedGeometries := []spatial.Geometry{}
+	for _, geom := range pf.geometries {
+		transformedGeometries = append(transformedGeometries, geom.Transform(transformByPose))
+	}
+	return NewGeometriesInFrame(pf.name, transformedGeometries), nil
+}
+
+// DoF returns the number of degrees of freedom within a model.
+func (pf *poseFrame) DoF() []Limit {
+	return pf.limits
+}
+
+// MarshalJSON serializes a Model.
+func (pf *poseFrame) MarshalJSON() ([]byte, error) {
+	return nil, errors.New("serializing a poseFrame is currently not supported")
+}
+
+// InputFromProtobuf converts pb.JointPosition to inputs.
+func (pf *poseFrame) InputFromProtobuf(jp *pb.JointPositions) []Input {
+	n := make([]Input, len(jp.Values))
```

Comment:
ok, I'll change it to be 
```
n[idx] = Input{d}
```

Reply:
I suppose we could have the last element convert since JointPositions are suppposed to be degrees but realistically this shouldnt matter as this wont be called in practice.

Reply:
> JointPositions are supposed to be degrees

Given this I would prefer to have this function (and the other places you commented regarding this) stay as degrees.

As you say, in practice it shouldn't be called so it won't matter. But in that case this function could just error.

But we may look at this a year from now and say "why did we do that?" and not know. Since we have a data structure that we have defined to be in a certain set of units, we should be consistent and treat it as those units.

Reply:
Yeah I agree its correct to make this change